shopinvader_payment/components/payment_transaction_event_listerner.py

```diff
@@ -26,14 +26,16 @@ def _confirm_and_invalidate_session(self, sale_order):
             # That should be the responsibility of the client
             # to request the cart info to be stored in its own cache
             # once the cache is reset.
-            collection = _PseudoCollection("shopinvader.backend", self.env)
-            work = self.work.work_on(collection=collection)
-            work.shopinvader_backend = shopinvader_backend
-            # another ugly hack: the partner here should be taken
-            # from current client request
-            work.partner = sale_order.partner_id
-            res = work.component(usage="cart")._to_json(sale_order)
-            response.set_store_cache("last_sale", res.get("data", {}))
+
+            invader_partner = sale_order.partner_id._get_invader_partner(
+                shopinvader_backend
+            )
```

Comment:
Hi,

I am using this code in production for few months.
Due to guess mode it may happen that the sale_order.partner_id has no active `shopinvader.partner` (because another guess partner with same email has been created meanwhile).
Thus invader_partner is empty and this code fails.

I think we should at least check that invader_partner is not empty.
Also I suggest to use `.with_context(active_test=False)` in order to avoid this issue.

```suggestion
            invader_partner = sale_order.partner_id.with_context(active_test=False)._get_invader_partner(
                shopinvader_backend
            )
            if not invader_partner:
                _logger.error(f"Could not find invader_partner for sale order {sale_order.id}")
                return
            if len(invader_partner) > 1:
                _logger.error(f"Several invader_partner found for sale order {sale_order.id}")
                return
            if not invader_partner.active:
                _logger.warning(f"Inactive invader_partner found for sale order {sale_order.id}")
```